homeassistant/components/mill/climate.py

```diff
@@ -117,20 +121,31 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         await self.coordinator.mill_data_connection.set_heater_temp(
             self._id, float(temperature)
         )
+        if hvac_mode := kwargs.get(ATTR_HVAC_MODE):
```

Comment:
While making code improvements such as this issue, consider adding type hints when possible. For example, here you could declare the variable before such as `hvac_mode: HVACMode | None = kwargs.get(ATTR_HVAC_MODE)`. Same goes for `temperature`.